site/sfguides/src/native-app-chairlift/native-app-chairlift.md

```diff
@@ -0,0 +1,632 @@
+
+
+<!-- ------------------------ -->
+## Set up account roles
+Duration: 3
+
+Let's start building the app. You'll first need to configure certain roles and permissions within your Snowflake account. This will allow you to view the app as an app admin (perhaps the chairlift manufacturer is doing some diagnostic work), or as an app viewer (perhaps the ski resort owner is interested in identifying chairlifts in need of repair).
+
+To create these roles and permissions, run the scripts below. You'll only need to execute these scripts once.
+
+**Execute prepare/provider-role.sql**
+
+Open a SQL worksheet in Snowsight and execute the following script:
+
+```sql
+-- create provider role
+create role if not exists chairlift_provider;
+grant role chairlift_provider to role accountadmin;
+grant create application package on account to role chairlift_provider;
+grant create database on account to role chairlift_provider;
+
+-- ensure a warehouse is usable by provider
+create warehouse if not exists chairlift_wh;
+grant usage on warehouse chairlift_wh to role chairlift_provider;
+```
+
+**Execute prepare/consumer-roles.sql**
+
+Open a SQL worksheet in Snowsight and execute the following script:
+
+```sql
+-- create consumer role
+create role if not exists chairlift_admin;
+create role if not exists chairlift_viewer;
+grant role chairlift_admin to role accountadmin;
+grant role chairlift_viewer to role accountadmin;
+grant create database on account to role chairlift_admin;
+grant create application on account to role chairlift_admin;
+grant execute task, execute managed task on account to role chairlift_admin with grant option;
+grant role chairlift_viewer to role chairlift_admin;
+
+-- ensure a warehouse is usable by consumer
+grant usage on warehouse chairlift_wh to role chairlift_admin;
+grant usage on warehouse chairlift_wh to role chairlift_viewer;
+```
+
+
+<!-- ------------------------ -->
+## Prepare objects in account
+Duration: 4
+
+Next, you'll run some scripts to setup some databases, schemas, and tables needed by the native app.
+
+The scripts will do a couple of things:
+
+* The provider scripts define roles, warehouses, and  data types for the data emitted by the sensors (i.e., brake temperature, motor RPMs, etc.). This is necessary so that the app can function as intended in the consumer's account. From this perspective, the provider is the chairlift manufacturer, and the native app will take raw sensor data from a consumer's account and generate chairlift analytics and insights for the consumer, granted that the consumer installs the app and grants the proper privileges to the app.
+
+* The consumer scripts mock fictional, raw sensor data from the chairlifts. From this perspective, the consumer is a purchaser of the chairlifts (i.e., owner of the ski resort) and has collected sensor data from the chairlifts since installation. The provider's app will run against this collected data in the customer's account.
```

Comment:
To elaborate, our mental model was the existence of a fictional second "connector" app that actually streams in this data. This is why it belongs in the consumer account. It also goes with the Snowflake ideology of customers owning their own data (i.e. it doesn't live in the provider's account, even though the provider is the one who is responsible for the machines that are generating the data).

Reply:
Scenario:

- Connector – we've mocked the data, in practice, this data would be provided by a connector. The data would be owned by the consumer, and be in that account. And then they could install the native app and use it to analyze the data collected via the connector.

- You could think of using these insights to setup appointments with the manufacturer (an extension).